src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4974,7 +4974,7 @@ internal override ReadOnlyPSMemberInfoCollection<T> Match(string name, PSMemberT
         /// <returns>The enumerator for this collection.</returns>
         public override IEnumerator<T> GetEnumerator()
         {
-            return new Enumerator<T>(this);
+            return new Enumerator(this);
```

Comment:
Why isn't it:
```suggestion
            return new Enumerator<T>(this);
```

Reply:
The type argument `S` is redundant and can be removed in: `PSMemberInfoIntegratingCollection<T>.Enumerator<S>` because it can be shown that S = T in all cases.